tests/integration/golang/auth/auth_oidc_based_test.go

```diff
@@ -0,0 +1,612 @@
+			AuthOIDCClaimRoles:       "groups",
+			AuthOIDCProviderEndpoint: oidcMockServer.Address(),
+		},
+	}
+	testSuite.oidcMockServer = oidcMockServer
+	suite.Run(t, testSuite)
+}
+
+func (s *OIDCAuthTestSuite) SetupTestSuite() {
+	// prepare test data.
+	namespace1, err := s.NamespaceFixtures.CreateNamespace(context.Background(), &models.Namespace{
+		ID:                  2,
+		Code:                "namespace1",
+		Description:         "Test namespace 1",
+		DefaultExperimentID: common.GetPointer(models.DefaultExperimentID),
+	})
+	s.Require().Nil(err)
+	s.namespace1 = namespace1
+
+	namespace2, err := s.NamespaceFixtures.CreateNamespace(context.Background(), &models.Namespace{
+		ID:                  3,
+		Code:                "namespace2",
+		Description:         "Test namespace 2",
+		DefaultExperimentID: common.GetPointer(models.DefaultExperimentID),
+	})
+	s.Require().Nil(err)
+	s.namespace2 = namespace2
+
+	namespace3, err := s.NamespaceFixtures.CreateNamespace(context.Background(), &models.Namespace{
+		ID:                  4,
+		Code:                "namespace3",
+		Description:         "Test namespace 3",
+		DefaultExperimentID: common.GetPointer(models.DefaultExperimentID),
+	})
+	s.Require().Nil(err)
+	s.namespace3 = namespace3
+
+	group1Role := models.Role{Name: "group1"}
+	s.Nil(s.RolesFixtures.CreateRole(context.Background(), &group1Role))
+	s.Nil(s.RolesFixtures.AttachNamespaceToRole(context.Background(), &group1Role, namespace1))
+
+	group2Role := models.Role{Name: "group2"}
+	s.Nil(s.RolesFixtures.CreateRole(context.Background(), &group2Role))
+	s.Nil(s.RolesFixtures.AttachNamespaceToRole(context.Background(), &group2Role, namespace2))
+	s.Nil(s.RolesFixtures.AttachNamespaceToRole(context.Background(), &group2Role, namespace3))
+
+	// create test users and obtain theirs tokens.
+	user1Token, err := s.oidcMockServer.Login(
+		context.Background(),
+		&mockoidc.MockUser{
+			Email:  "test.user@example.com",
+			Groups: []string{"group1"},
+		}, []string{"openid", "groups"},
+	)
+	s.Nil(err)
+	s.user1Token = user1Token
+
+	user2Token, err := s.oidcMockServer.Login(
+		context.Background(),
+		&mockoidc.MockUser{
+			Email:  "test.user@example.com",
+			Groups: []string{"group2"},
+		}, []string{"openid", "groups"},
+	)
+	s.Nil(err)
+	s.user2Token = user2Token
+
+	user3Token, err := s.oidcMockServer.Login(
+		context.Background(),
+		&mockoidc.MockUser{
+			Email:  "test.user@example.com",
+			Groups: []string{"group1", "group2"},
+		}, []string{"openid", "groups"},
+	)
+	s.Nil(err)
+	s.user3Token = user3Token
+
+	user4Token, err := s.oidcMockServer.Login(
+		context.Background(),
+		&mockoidc.MockUser{
+			Email:  "test.user@example.com",
+			Groups: []string{"admin"},
+		}, []string{"openid", "groups"},
+	)
+	s.Nil(err)
+	s.user4Token = user4Token
+}
+
+func (s *OIDCAuthTestSuite) TestAIMAuth_Ok() {
+	s.SetupTestSuite()
+
+	tests := []struct {
+		name  string
+		check func()
+	}{
+		{
+			name: "TestUser1Access",
```

Comment:
Maybe better to name these for different aspects of test, eg "UserCannotAccessUnauthorizedNamespace"